src/edu/washington/escience/myria/operator/DataOutput.java

```diff
@@ -61,6 +69,9 @@ protected void consumeTuples(final TupleBatch tuples) throws DbException {
   @Override
   protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbException {
     try {
+      if (dataSink != null) {
+        tupleWriter.open(dataSink.getOutputStream());
+      }
       tupleWriter.writeColumnHeaders(getChild().getSchema().getColumnNames());
```

Comment:
I completely agree that a new `DataSink` impl for downloading data is the way to go!